### DISCUSSION TITLE
How to cancel/abort a workflow?

### DISCUSSION THREAD

**Discussion by enaoki** [elsa-workflows/elsa-core#1093] [Q&A] [answered]
Maybe similar to [this question](https://github.com/elsa-workflows/elsa-core/discussions/358) but I didn't want to mix the issues.
But how could we do it? Is it enough just updating the workflow status to `Cancelled`? Or do we need to delete `Bookmark` records, `BlockingActivities` field in `WorkflowInstance` table as well?

**Comment by sfmskywalker** [comment, marked as answer]
Technically, it's sufficient to just set the status to `Cancelled` like you said. But to keep things tidy, maybe it is a good idea to also clean the `Bookmarks` table as well. And perhaps indeed also the `BlockingActivities`, `ScheduledActivities`, `Scopes` and nullify the `CurrentActivity` field. But not doing this doesn't hurt.
